Xamarin.Essentials/Preferences/Preferences.win32.cs

```diff
@@ -1,43 +1,63 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
-using Newtonsoft.Json;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Formatters.Binary;
 
 namespace Xamarin.Essentials
 {
     public static partial class Preferences
     {
         static readonly string settingsPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), AppInfo.Name, "settings.dat");
 
```

Comment:
Any chance that AppInfo.Name will contain invalid chars for win32 file names, such as a ":" ?

Reply:
Good catch, the documentation for the attribute says "The assembly title is a friendly name, which can include spaces" but that's not particularly informative.

Reply:
It could also be blank ... perhaps fallback to using the file name of the launching assembly if the sanitized name is blank?

``` c#
      var name = System.IO.Path.GetFileNameWithoutExtension(Assembly.GetEntryAssembly().CodeBase);
```